src/coreclr/vm/jithelpers.cpp

```diff
@@ -2105,48 +2105,43 @@ HCIMPLEND
 //*************************************************************
 // Allocation fast path for typical objects
 //
-HCIMPL1(Object*, JIT_NewS_MP_FastPortable, CORINFO_CLASS_HANDLE typeHnd_)
+HCIMPL1_RAW(Object*, JIT_NewS_MP_FastPortable, CORINFO_CLASS_HANDLE typeHnd_)
```

Comment:
What's the problem with using regular HCMPL1 here?

Reply:
The FCall prolog inserts an unreachable `setjmp` on Win-x64 to disable some optimizations. As a result, MSVC refuses to produce a tail-call to the `JIT_New` slow helper (I verified in MSVC's source that the `setjmp` makes their tail-call optimization illegal). By using the `_RAW` variants, we don't insert the prolog and MSVC produces the tail call.

Reply:
Ok, makes sense. I can't wait for all HELPER_METHOD_FRAME to disappear and the setjmp hack with them.